runtime/lua/vim/inspect.lua

```diff
@@ -1,3 +1,9 @@
+--- @class vim.inspect.Opts
+--- @field depth? integer
+--- @field newline? string
+--- @field process? fun(item:any, path: string[]): any
+
+--- @overload fun(x: any, opts?: vim.inspect.Opts): string
```

Comment:
Nice, I also once needed this! But in order to make this bundled file as-is (although we already have some neovim-specific modifications), how about adding the annotations to `vim/_editor.lua`?

L199: `vim.inspect = vim.inspect`

Reply:
We have quite a few Neovim specific modifications to inspect.lua so I think we basically own that file now.

I can move the annotations anyway.

Reply:
Yes, upstream is in teal (🤢) and bfredl made a number of changes to support inspecting Vim types (like `vim.NIL`). All our inspect belong to us now.

Reply:
done